src/components/All_Articles/list_of_articles.jsx

```diff
@@ -1,24 +1,18 @@
-import getUsers from "../../api";
-import { useState, useEffect } from "react";
 import {Link} from 'react-router-dom'
-const Articles = () => {
 
-    const [articles, setArticles] = useState([])
-
-    useEffect(() => {
-        getUsers().then((articles => {
-            setArticles(articles)
-        }))
-    }, [])
+const ListOfArticles = ({articles, setArticle}) => {
 
+    function handleClick(articleId){
+        setArticle(articleId)
+    }
     return(
         <section>
             <h2 className="heading">Articles</h2>
             <ul className="articlesList">
                 {articles.map((article) => {
                     return <li id="articlesArticleCards" key={article.article_id}>  
                         <section id="articleCard">
-                            <Link to="/article" id='articleLink'><h4 className="heading">{article.title}</h4></Link>
+                            <Link onClick={() => {handleClick(article.article_id)}} to="/article" id='articleLink'><h4 className="heading">{article.title}</h4></Link>
```

Comment:
Links take paths not onClick functions. Link already has an onClick designed to navigate the user between pages. This is not an appropriate use of this functionality.